spark/interpreter/src/main/java/org/apache/zeppelin/spark/Utils.java

```diff
@@ -149,7 +149,7 @@ static boolean isSpark2() {
   }
   
   public static String buildJobGroupId(InterpreterContext context) {
-    return "zeppelin-" + context.getNoteId() + "-" + context.getParagraphId();
+    return context.getAuthenticationInfo().getUser() + "-" + context.getNoteId() + "-" + context.getParagraphId();
```

Comment:
Is it possible `context.getAuthenticationInfo()` is null or `context.getAuthenticationInfo().getUser()` is null